### PR TITLE
Cleanup some of the internal API usage in Relational

### PR DESCRIPTION
**Pull request by smitpatel**
- Make EnumerableExtensions shared internal
- Make ExpressionEqualityComparer public
- Suppress Internal usage in EntityFrameworkRelationalServicesBuilder
- Make ConfigurationSourceExtensions public
- Suppress convention-less modelBuilder creation in Migration/Snapshot

### CODE REVIEW

File: src/EFCore/Extensions/EntityTypeExtensions.cs [hunk 1]
```diff
@@ -48,6 +48,9 @@ public static IEntityType GetRootType([NotNull] this IEntityType entityType)
             return entityType.BaseType?.GetRootType() ?? entityType;
         }
 
+        public static IEnumerable<IEntityType> GetAllBaseTypes([NotNull] this IEntityType entityType)
```
Comment (AndriySvyryd) [inline src/EFCore/Extensions/EntityTypeExtensions.cs]: Docs for all of these

Comment (smitpatel) [reply src/EFCore/Extensions/EntityTypeExtensions.cs]: I should have just filed new issue. Some of these methods even I don't understand hat they do. Like IsInOwnershipPath. 😛

Comment (AndriySvyryd) [reply src/EFCore/Extensions/EntityTypeExtensions.cs]: "Determines whether the given entity type is in the ownership path" 😆

Comment (smitpatel) [reply src/EFCore/Extensions/EntityTypeExtensions.cs]: Updated.

File: src/EFCore/Extensions/MutableEntityTypeExtensions.cs [hunk 2]
```diff
@@ -54,6 +65,14 @@ public static IEnumerable<IMutableEntityType> GetDerivedTypesInclusive([NotNull]
         public static IEnumerable<IMutableEntityType> GetDirectlyDerivedTypes([NotNull] this IMutableEntityType entityType)
             => ((EntityType)entityType).GetDirectlyDerivedTypes();
 
+        /// <summary>
+        ///     Returns all types in hierarchy of the given <see cref="IMutableEntityType" />.
```
Comment (AndriySvyryd) [inline src/EFCore/Extensions/MutableEntityTypeExtensions.cs]: Technically it only returns all base and derived types (no siblings)

**Pull request merged**